### PR TITLE
Play videos time-dependent

### PR DESCRIPTION
**Pull request by faizrr**
This feature allow screensaver play day videos only during the day (from 6.00 to 20.00) and play night videos only during the night (at other time)

### CODE REVIEW

**Comment by mateusrevoredo** [comment]
Awesome @faizrr! That's the first thing I thought when I downloaded this Screensaver. I was thinking "What if we could set up a preference to make the videos time sensitive..."

I was planning to do that on the weekend and submit a PR, when I decided to look at the pending ones. 
Thankfully someone had already submitted a PR for this.Hope it gets merged.

**Comment by kerstin** [comment]
Awesome^2!

**Comment by aimonb** [comment]
Huge +1 on adding this feature.

File: Aerial/Source/Models/ManifestLoader.swift [hunk 1]
```diff
@@ -50,6 +51,25 @@ class ManifestLoader {
                 }
             }
             
+            if timeDependent == true {
```
Comment (LarsStegman) [inline Aerial/Source/Models/ManifestLoader.swift]: Couldn't you just remove the '== true'? It's unnecessary.

File: Aerial/Source/Models/ManifestLoader.swift [hunk 2]
```diff
@@ -50,6 +51,25 @@ class ManifestLoader {
...
+                let date = NSDate()
+                let calendar = NSCalendar.currentCalendar()
+                let components = calendar.components(NSCalendarUnit.Year.union(NSCalendarUnit.Hour),fromDate: date)
+                let hour = components.hour
+                
+                if hour > 6 && hour < 20 {
```
Comment (LarsStegman) [inline Aerial/Source/Models/ManifestLoader.swift]: Maybe make these boundaries a preference?

**Comment by nuno-agostinho** [comment]
A bit more complex but cooler: adding a way to know the day's sunset/sunrise time of the user's place (for example, using external APIs like the [Wolfram Alpha's](http://products.wolframalpha.com/api)). That'd be better than selecting specific hours since countries/seasons have different daylight duration.

An extra bonus: if the screensaver activates near the sunrise/sunset time, it could show one of the sunsets or sunrises available! But as they are of low duration, maybe it isn't such great idea… :(

**Comment by steve1412** [comment]
I didn't know this was already raised as feature suggestion. count me in +1

P.S Flux has a GPS which detects when the sunset happens. Perhaps is something that can be integrated in the video. It doesn't have to be sunset, but perhaps more accurate by differentiating night time or afternoon, or if it's cloudy or foggy (more of a weather thing now, but this would hugely depend on the source available from Apple)

**Comment by geo4orce** [comment]
+1

**Comment by jaggs6** [comment]
+1

**Comment by glouel** [comment]
So right now this no longer works with the current JSON, see comment here 
https://github.com/JohnCoates/Aerial/pull/317

**Comment by glouel** [comment]
This has been reimplemented for 1.4 as the new JSONs don't provide the night/day information anymore. Thanks for the contribution!

**Pull request closed without merging**